f3/granite.go

```diff
@@ -921,6 +926,41 @@ func (q *quorumState) getSigners(value ECChain) bitfield.BitField {
 	return signers
 }
 
+// getStrongQuorumSigners retrieves just a strong quorum of signers of the given ECChain.
+// At the moment, this is the signers with the most power until reaching a strong quorum.
+func (q *quorumState) getStrongQuorumSigners(value ECChain) (bitfield.BitField, error) {
+	signers := q.getSigners(value)
+	strongQuorumSigners := bitfield.New()
+	justificationPower := NewStoragePower(0)
+	err := signers.ForEach(func(bit uint64) error {
+		if int(bit) >= len(q.powerTable.Entries) {
+			return fmt.Errorf("invalid signer index: %d", bit)
+		}
+		justificationPower.Add(justificationPower, q.powerTable.Entries[bit].Power)
+		strongQuorumSigners.Set(bit)
+		if hasStrongQuorum(justificationPower, q.powerTable.Total) {
+			return StrongQuorumReachedError{} // No need to iterate anymore
+		}
+		return nil
+	})
+
+	if _, ok := err.(StrongQuorumReachedError); ok {
+		// Strong quorum reached (not an actual error)
+		return strongQuorumSigners, nil
+	}
+
+	// if we didn't find a strong quorum,
+	// return an empty bitfield, and whatever
+	// error that was returned by ForEach (if any)
+	return bitfield.New(), err
+}
+
+type StrongQuorumReachedError struct{}
```

Comment:
You can use errors.New, defined as a var, and then errors.Is to detect it.

Reply:
Done.